build-system/babel-config/pre-closure-config.js

```diff
@@ -0,0 +1,108 @@
+  const filterImportsPlugin = [
+    'filter-imports',
+    {
+      imports: {
+        // Imports removed for all ESM builds.
+        './polyfills/document-contains': ['installDocContains'],
+        './polyfills/domtokenlist': ['installDOMTokenList'],
+        './polyfills/fetch': ['installFetch'],
+        './polyfills/math-sign': ['installMathSign'],
+        './polyfills/object-assign': ['installObjectAssign'],
+        './polyfills/object-values': ['installObjectValues'],
+        './polyfills/promise': ['installPromise'],
+        './polyfills/array-includes': ['installArrayIncludes'],
+        './ie-media-bug': ['ieMediaCheckAndFix'],
+        '../third_party/css-escape/css-escape': ['cssEscape'],
+        // Imports that are not needed for valid transformed documents.
+        '../build/ampshared.css': ['cssText', 'ampSharedCss'],
+        '../build/ampdoc.css': ['cssText', 'ampDocCss'],
+      },
+    },
+  ];
+  const reactJsxPlugin = [
+    '@babel/plugin-transform-react-jsx',
+    {
+      pragma: 'Preact.createElement',
+      pragmaFrag: 'Preact.Fragment',
+      useSpread: true,
+    },
+  ];
+
+  const preClosurePlugins = [
+    './build-system/babel-plugins/babel-plugin-transform-fix-leading-comments',
+    '@babel/plugin-transform-react-constant-elements',
+    reactJsxPlugin,
+    './build-system/babel-plugins/babel-plugin-transform-inline-configure-component',
+    // TODO(alanorozco): Remove `replaceCallArguments` once serving infra is up.
+    [
+      './build-system/babel-plugins/babel-plugin-transform-log-methods',
+      {replaceCallArguments: false},
+    ],
+    './build-system/babel-plugins/babel-plugin-transform-parenthesize-expression',
+    './build-system/babel-plugins/babel-plugin-is_minified-constant-transformer',
+    './build-system/babel-plugins/babel-plugin-transform-amp-extension-call',
+    './build-system/babel-plugins/babel-plugin-transform-html-template',
+    './build-system/babel-plugins/babel-plugin-transform-version-call',
+    './build-system/babel-plugins/babel-plugin-transform-simple-array-destructure',
+    replacePlugin,
+    argv.single_pass
+      ? './build-system/babel-plugins/babel-plugin-transform-amp-asserts'
+      : null,
+    argv.esm ? filterImportsPlugin : null,
+    argv.esm
+      ? './build-system/babel-plugins/babel-plugin-transform-function-declarations'
+      : null,
+    isCheckTypes
+      ? './build-system/babel-plugins/babel-plugin-transform-simple-object-destructure'
+      : './build-system/babel-plugins/babel-plugin-transform-json-configuration',
+    !(argv.fortesting || isCheckTypes)
+      ? [
+          './build-system/babel-plugins/babel-plugin-amp-mode-transformer',
+          {isEsmBuild: !!argv.esm},
+        ]
+      : null,
+    !(argv.fortesting || isCheckTypes)
+      ? './build-system/babel-plugins/babel-plugin-is_dev-constant-transformer'
+      : null,
+    argv.single_pass ? 'transform-es2015-modules-commonjs' : null,
```

Comment:
This is only used when building the dep tree, not for the regular compile. That means we need a special `single-pass-deps` config for building the deps, and we'll use the regular `pre-closure` config when compiling in single pass.

Reply:
Good catch! Done.